rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -2792,17 +2795,17 @@ int LuaSyncedCtrl::SetProjectileTarget(lua_State* L)
 	switch (lua_gettop(L)) {
 		case 3: {
 			const int id = luaL_checkint(L, 2);
-			const int type = luaL_checkint(L, 3);
+			const string type = luaL_checkstring(L, 3);
 
 			CWorldObject* oldTargetObject = wpro->GetTargetObject();
 			CWorldObject* newTargetObject = NULL;
 
-			switch (type) {
```

Comment:
whats wrong with switch/case here? imo switch/case should be prefered over if/else when possible.